mlrun/api/crud/model_monitoring/stores/sql_model_endpoint_store.py

```diff
@@ -0,0 +1,571 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import typing
+
+import pandas as pd
+import sqlalchemy as db
+
+import mlrun
+import mlrun.api.schemas
+import mlrun.model_monitoring.constants as model_monitoring_constants
+import mlrun.utils.model_monitoring
+import mlrun.utils.v3io_clients
+from mlrun.api.db.sqldb.session import DBEngine, create_session
+from mlrun.utils import logger
+
+from .model_endpoint_store import ModelEndpointStore
+
+
+class SQLModelEndpointStore(ModelEndpointStore):
+
+    """
+    Handles the DB operations when the DB target is from type SQL. For the SQL operations, we use SQLAlchemy, a Python
+    SQL toolkit that handles the communication with the database.  When using SQL for storing the model endpoints
+    record, the user needs to provide a valid connection string for the database.
+    """
+
+    _engine = None
+
+    def __init__(
+        self,
+        project: str,
+        connection_string: str = None,
+    ):
+        """
+        Initialize SQL store target object.
+
+        :param project: The name of the project.
+        :param connection_string: Valid connection string or a path to SQL database with model endpoints table.
+        """
+
+        super().__init__(project=project)
+        self.connection_string = connection_string
+        self.table_name = model_monitoring_constants.EventFieldType.MODEL_ENDPOINTS
+        self.metadata = db.MetaData()
+        self._db_connection = DBEngine(dsn=connection_string)
+        self.init_engine()
```

Comment:
No need for all this - the `DBEngine` class should not be passed here, it's just a utility to maintain the cache of DB engines. All you need to call here is the global `create_session()` but provide a `dsn` - you'll get back a DB session ready to work with.
In the initialization step, you'll still need to verify the table exists, of course.